docs/content/en/reference/native-extensions.md

```diff
@@ -29,6 +29,10 @@ Currently stencil does not provide a testing framework for native extensions, bu
 
 A native extension can be ran locally using the `replacements` key in an application's manifest (`service.yaml`), which is described in the [module documentation]({{< relref "template-module#testing-a-module-used-in-a-stencil-application" >}}). However, when doing this the native extension must write it's binary to `bin/plugin`.
 
+## Debugging a Native Extension
+
+The [`go-plugin`](https://github.com/hashicorp/go-plugin) library does not surface errors to stencil. Instead, it will raise the generic message `failed to create connection to extension: Unrecognized remote plugin message`. To determine a more precise error message, execute the native extension binary directly.
```

Comment:
It might've been worth noting where the native extension binaries live, or how to find it in the error.

Reply:
Make sense. I will create another PR to add the notes.